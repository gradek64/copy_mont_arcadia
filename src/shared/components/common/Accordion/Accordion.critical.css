/** @define Accordion */

.Accordion {

  &-wrapper {
    transition: max-height $accordion-anim-time;
    overflow: hidden;
  }

  &-content {
    visibility: hidden;
    transition: visibility 0s linear $accordion-anim-time;

    &.is-padded {
      padding: 0 $page-padding calc(5px + $page-padding) $page-padding;
    }

    &--borderTop {
      border-top: $accordion-content-border-top;
    }
  }

  /* Expanded state */
  &.is-expanded {
    & .Accordion {
      &-icon {
        &--primary {
          @if $accordion-icon-primary-is-plus == false {
            transform: rotate($accordion-icon-primary-rotation-expanded);
          }
        }

        &--secondary {
          @if $accordion-icon-secondary-is-plus == false {
            transform: rotate($accordion-icon-secondary-rotation-expanded);
          }
        }

        &::before,
        &::after {
          transform: rotate(0);
        }
      }

      &-header {
        color: $accordion-expanded-header-font-color;
        background-color: $accordion-expanded-header-background-color;
      }

      &-content {
        &.is-visible {
          visibility: visible;
        }
        transition: visibility 0s linear 0s;
      }
    }
  }

}
