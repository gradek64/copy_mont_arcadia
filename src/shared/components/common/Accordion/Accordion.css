/** @define Accordion */

.Accordion {
  text-align: left;
  transition: height $accordion-anim-time ease-in-out;
  border-width: $accordion-border-width;
  border-style: solid;
  border-color: $accordion-border-color;

  &-withoutBorders {
    border:none
  }

  &-header {
    display: flex;
    align-items: $accordion-header-align-items;
    position: relative;
    padding: $accordion-header-padding 0;
    cursor: pointer;
    -webkit-tap-highlight-color: transparent;
    color: $accordion-header-font-color;
    background-color: $accordion-header-background-color;
    transition: background-color $accordion-anim-time linear;

    &.is-padded {
      padding-left: $page-padding;
      padding-right: $page-padding;
    }
  }

  &-title {
    flex: 1;
    overflow: hidden;
    user-select: none;
    min-height: inherit;
    text-transform: $accordion-title-text-transform;
  }

  &-icon {
    position: relative;
    align-self: center;
    transition: transform $accordion-anim-time ease-in-out;
    background: none no-repeat center/contain transparent;

    &--hidden {
      visibility: hidden;
    }

    &--left {
      margin-right: 10px;
    }

    &--right {
      order: 1;
      margin-left: 10px;
    }

    &::before,
    &::after {
      content: '';
      position: absolute;
      display: none;
      width: 100%;
      top: 50%;
      margin-top: calc(($accordion-icon-primary-plus-thickness / 2) * -1);
      transition: transform $accordion-anim-time ease-in-out;
    }

    &::before {
      transform: rotate(-180deg);
    }

    &::after {
      transform: rotate(-90deg);
    }

    &--primary {
      height: $accordion-icon-primary-size;
      flex: 0 0 $accordion-icon-primary-size;

      @if $accordion-icon-primary-is-plus {
        background: $accordion-icon-primary-plus-background-color;
        border-radius: 50%;

        &::before,
        &::after {
          display: block;
          height: $accordion-icon-primary-plus-thickness;
          background: $accordion-icon-primary-plus-color;

          @if $accordion-icon-primary-plus-background-color != null {
            left: 25%;
            width: 50%;
            margin-top: calc(($accordion-icon-primary-plus-thickness / 2) * -1);
          }
        }
      } @else {
        background-image: url('public/$brand/images/accordion-icon-primary-down.svg');
        transform: rotate($accordion-icon-primary-rotation-collapsed);
      }
    }

    &--secondary {
      height: $accordion-icon-secondary-size;
      flex: 0 0 $accordion-icon-secondary-size;

      @if $accordion-icon-secondary-is-plus {
        background: $accordion-icon-secondary-plus-background-color;
        border-radius: 50%;

        &::before,
        &::after {
          display: block;
          height: $accordion-icon-secondary-plus-thickness;
          background: $accordion-icon-secondary-plus-color;

          @if $accordion-icon-secondary-plus-background-color != null {
            left: 25%;
            width: 50%;
            margin-top: calc(($accordion-icon-primary-plus-thickness / 2) * -1);
          }
        }
      } @else {
        background-image: url('public/$brand/images/accordion-icon-secondary-down.svg');
        transform: rotate($accordion-icon-secondary-rotation-collapsed);
      }
    }
  }

  &-statusIndicatorText {
    position: relative;
    order: 1;
    margin-left: 10px;

    &--hidden {
      visibility: hidden;
    }
  }

  &-wrapper {
    transition: max-height $accordion-anim-time;
    overflow: hidden;
  }

  &.is-expanded .Accordion-header--noExpandedBackground {
    background-color: $accordion-header-background-color;
  }

  &--noBorders {
    border: none;
  }

  &--swatches {
    margin: 0;
    border: none;
    &.is-expanded {
      & .Accordion-header {
        background-color: $accordion-header-background-color;
      }
    }
    & > .Accordion-wrapper {
      margin-top: -15px;
      & > .Accordion-content {
        border: none;
        &.is-padded {
          padding: 0 $accordion-header-padding;
        }
      }
    }
  }

  /* Members of an accordion group do not get a bottom border to avoid double borders */
  &-groupMember:not(:last-child) {
    border-bottom-width: 0;
  }

  &--storeLocator {
    background-color: $store-locator-accordion-background-color;
    border-width: $store-locator-accordion-border-width;
    &.is-expanded {
      & > .Accordion-header {
        background-color: $store-locator-accordion-background-color;
      }
      & > .Accordion-wrapper {
        & > .Accordion-content {
          border-top: $store-locator-accordion-wrapper-border-top;
        }
      }
    }
  }

  &--checkoutDeliveryTypeAccordion {
    background-color: $checkout-delivery-type-accordion-background;
    &.is-expanded {
      & > .Accordion-header {
        background-color: unset;
        & > .Accordion-icon {
          display: none;
        }
      }
    }
    & > .Accordion-header {
      display: block;
      padding-top: 0;
      & > .Accordion-statusIndicatorText {
        position: absolute;
        right: 0;
        font-size: $delivery-type-statusIndicatorText-font-size;
        letter-spacing: $delivery-type-statusIndicatorText-letter-spacing;
        border-bottom: $delivery-type-statusIndicatorText-border-bottom;
        padding-bottom: $delivery-type-statusIndicatorText-padding-bottom;
        height: $delivery-type-statusIndicatorText-height;
        /* postcss-bem-linter: ignore */
        a {
          text-decoration: none;
        }
      }
      & > .Accordion-title {
        /* postcss-bem-linter: ignore */
        & div {
          line-height: $checkout-delivery-type-title-line-height;
          margin-bottom: $checkout-delivery-type-title-margin-bottom;
        }
      }

      & > .Accordion-subTitle {
        display: block;
      }
    }
  }

  &--deliveryInstructions {
    border: 0;

   & .Accordion-header {
        padding-bottom: 3px;
     }

    & .Accordion-title {
      font-size: 16px;
      line-height: 20px;
      flex: initial;
    }

    & .Accordion-content {
      padding-top: 10px;
      border: 0;
    }
  }
}

