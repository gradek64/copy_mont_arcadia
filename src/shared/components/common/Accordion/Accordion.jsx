import classnames from 'classnames'
import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { path, equals } from 'ramda'
import keys from '../../../constants/keyboardKeys'
import Loader from '../Loader/Loader'

export default class Accordion extends Component {
  static propTypes = {
    accordionName: PropTypes.string,
    expanded: PropTypes.bool,
    showLoader: PropTypes.bool,
    header: PropTypes.node.isRequired,
    children: PropTypes.node,
    className: PropTypes.string,
    noContentPadding: PropTypes.bool,
    noContentBorderTop: PropTypes.bool,
    noHeaderPadding: PropTypes.bool,
    noMaxHeight: PropTypes.bool,
    onAccordionToggle: PropTypes.func,
    noExpandedHeaderBackground: PropTypes.bool,
    arrowStyle: PropTypes.oneOf(['primary', 'secondary']),
    arrowPosition: PropTypes.oneOf(['left', 'right', 'hidden']),
    subHeader: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
    withoutBorders: PropTypes.bool,
    statusIndicatorText: PropTypes.string,
    isDisabled: PropTypes.bool,
  }

  static defaultProps = {
    accordionName: '',
    expanded: false,
    showLoader: false,
    className: '',
    noContentBorderTop: false,
    noExpandedHeaderBackground: false,
    noMaxHeight: false,
    arrowStyle: 'primary',
    arrowPosition: 'left',
    children: null,
    onAccordionToggle: () => {},
    subHeader: '',
    withoutBorders: false,
    statusIndicatorText: '',
    isDisabled: false,
  }

  constructor(props) {
    super(props)
    this.state = {
      expanded: props.expanded,
      expandedHeight: 0,
      hasCollapseAnimationFinished: !props.expanded,
    }
  }

  UNSAFE_componentWillReceiveProps(nextProps) {
    if (nextProps.expanded !== this.state.expanded) {
      this.setState({ expanded: nextProps.expanded })
    }
  }

  shouldComponentUpdate(nextProps, nextState) {
    return (
      this.state.expanded !== nextState.expanded ||
      this.state.expandedHeight !== nextState.expandedHeight ||
      !equals(this.props.children, nextProps.children) ||
      (this.props.showLoader && !nextProps.showLoader) ||
      !equals(this.props.className, nextProps.className)
    )
  }

  componentDidMount() {
    this.updateExpandedHeight()
  }

  componentDidUpdate() {
    this.updateExpandedHeight()
  }

  updateExpandedHeight = () => {
    const newHeight = this.props.noMaxHeight
      ? 'none'
      : path(['scrollHeight'], this.accordionWrapper)

    if (newHeight === undefined) {
      return
    }

    if (newHeight !== this.state.expandedHeight) {
      this.setState({ expandedHeight: newHeight })
    }
  }

  toggle = (ev) => {
    if (ev) {
      ev.preventDefault()
    }

    const {
      accordionName,
      children,
      onAccordionToggle,
      isDisabled,
    } = this.props
    const notExpanded = !this.state.expanded

    if (!children || isDisabled) {
      return null
    }

    if (notExpanded) {
      this.setState({ hasCollapseAnimationFinished: false })
    }

    onAccordionToggle(accordionName, notExpanded)

    this.setState((state) => ({ expanded: !state.expanded }))
  }

  onKeyDown = ({ keyCode }) => {
    if (keyCode === keys.ENTER || keyCode === keys.SPACE) {
      this.toggle()
    }
  }

  handleOnTransactionEnd = () => {
    if (!this.props.expanded)
      this.setState({ hasCollapseAnimationFinished: true })
  }

  render() {
    const {
      children,
      className,
      header,
      noContentPadding,
      noHeaderPadding,
      noContentBorderTop,
      noExpandedHeaderBackground,
      arrowStyle,
      arrowPosition,
      showLoader,
      subHeader,
      statusIndicatorText,
      withoutBorders,
      isDisabled,
    } = this.props
    const { expanded } = this.state
    const classNames = classnames('Accordion', className, {
      'Accordion-withoutBorders': withoutBorders,
      'is-expanded': expanded,
    })
    const headerClassNames = classnames(Accordion.headerClassName, {
      [`${className}-headerContainer`]: !!className,
      'Accordion-header--noExpandedBackground': noExpandedHeaderBackground,
      'is-padded': !noHeaderPadding,
    })
    const iconClassNames = classnames(
      'Accordion-icon',
      `Accordion-icon--${arrowStyle}`,
      `Accordion-icon--${arrowPosition}`,
      {
        'Accordion-icon--hidden': !children || isDisabled,
      }
    )
    const statusIndicatorTextClassNames = classnames(
      'Accordion-statusIndicatorText',
      {
        'Accordion-statusIndicatorText--hidden': !children || isDisabled,
      }
    )
    const ctaClassNames = statusIndicatorText
      ? statusIndicatorTextClassNames
      : iconClassNames

    const childrenWithProps =
      children &&
      React.Children.map(
        children,
        (child) =>
          child &&
          React.cloneElement(child, {
            hidden: !expanded && this.state.hasCollapseAnimationFinished,
          })
      )

    return (
      <article
        className={classNames}
        ref={(accordion) => {
          this.accordion = accordion
        }}
      >
        <div // eslint-disable-line jsx-a11y/no-static-element-interactions
          className={headerClassNames}
          onClick={this.toggle}
          onKeyDown={this.onKeyDown}
          style={!isDisabled && children ? {} : { cursor: 'auto' }}
        >
          <div className={ctaClassNames}>
            {statusIndicatorText ? <a href="#">{statusIndicatorText}</a> : ''}
          </div>
          <header
            className="Accordion-title"
            role="button"
            tabIndex="0"
            aria-pressed={expanded}
          >
            {header}
          </header>
          {subHeader && <span className="Accordion-subTitle">{subHeader}</span>}
        </div>
        <div
          className="Accordion-wrapper"
          onTransitionEnd={() => this.handleOnTransactionEnd()}
          style={{ maxHeight: expanded ? this.state.expandedHeight : 0 }}
          ref={(accordionWrapper) => {
            this.accordionWrapper = accordionWrapper
          }}
        >
          {showLoader && <Loader />}
          {children && (
            <div
              className={classnames(`Accordion-content`, {
                [`${className}-content`]: !!className,
                'is-padded': !noContentPadding,
                'is-visible': !showLoader,
                'Accordion-content--borderTop': !noContentBorderTop,
              })}
            >
              {typeof children === 'object' ? childrenWithProps : children}
            </div>
          )}
        </div>
      </article>
    )
  }
}

Accordion.headerClassName = 'Accordion-header'
