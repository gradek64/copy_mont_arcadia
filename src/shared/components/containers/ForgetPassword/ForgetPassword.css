/** @define ForgetPassword */

.ForgetPassword {
  margin-top: 10px;

  &-form {
    padding: 0 $page-padding 10px;
  }
}

@media $min-tablet {
  .ForgetPassword {
    &--signin {
      padding: 0 $page-padding;
      & .ForgetPassword {
        &-form {
          padding: 0;
        }
      }
    }
  }
}

@media $min-laptop {
  .ForgetPassword {
    &--signin {
      & .ForgetPassword {
        &-accordion {
          width: 100%;
        }
      }
    }
  }
}
