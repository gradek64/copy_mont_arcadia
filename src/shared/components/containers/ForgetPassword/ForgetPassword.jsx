import React, { Component } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import { isFeatureDesktopResetPasswordEnabled } from '../../../selectors/featureSelectors'
import {
  forgetPwdRequest,
  resetPasswordLinkRequest,
  setForgetPassword,
} from '../../../actions/common/accountActions'
import {
  resetForm,
  setFormField,
  setFormMessage,
  touchedFormField,
} from '../../../actions/common/formActions'
import Accordion from '../../common/Accordion/Accordion'
import ForgetPasswordForm from '../../common/ForgetPasswordForm/ForgetPasswordForm'

class ForgetPassword extends Component {
  constructor(props) {
    super(props)
    this.scrollRef = null
  }
  static propTypes = {
    className: PropTypes.string,
    forgetPasswordForm: PropTypes.object,
    forgetPwdRequest: PropTypes.func,
    isFeatureDesktopResetPasswordEnabled: PropTypes.bool,
    resetForm: PropTypes.func,
    resetPasswordLinkRequest: PropTypes.func,
    setForgetPassword: PropTypes.func,
    setFormField: PropTypes.func,
    setFormMessage: PropTypes.func,
    touchedFormField: PropTypes.func,
    orderId: PropTypes.number,
    enableScrollToMessage: PropTypes.bool,
    noContentBorderTop: PropTypes.bool,
  }

  static defaultProps = {
    className: '',
    enableScrollToMessage: false,
    noContentBorderTop: false,
  }

  static contextTypes = {
    l: PropTypes.func,
  }

  state = {
    isAccordionExpanded: false,
  }

  onAccordionToggle = () => {
    this.setState((state) => ({
      isAccordionExpanded: !state.isAccordionExpanded,
    }))
  }

  clickedInAccordionTitle = (e) => {
    const titleEl = this.scrollRef.getElementsByClassName(
      Accordion.headerClassName
    )[0]

    if (!titleEl)
      throw new Error('Missing Accordion.headerClassName className!')

    const clickedInTitle =
      titleEl.getElementsByClassName(e.target.className).length > 0
    const clickTitle = titleEl === e.target

    return clickTitle || clickedInTitle
  }

  scrollToMyRef = (e) => {
    if (!this.clickedInAccordionTitle(e) || this.state.isAccordionExpanded)
      return

    const eleCoords = this.scrollRef.getBoundingClientRect()
    const scrollY = window.scrollY
    const pos = eleCoords.top + scrollY - eleCoords.height

    window.scroll({
      top: pos,
      behavior: 'smooth',
    })
  }

  render() {
    const { l } = this.context
    const { className, noContentBorderTop, ...props } = this.props
    return (
      // eslint-disable-next-line jsx-a11y/no-static-element-interactions
      <section
        className={`ForgetPassword ${className}`}
        role="presentation"
        ref={(ref) => {
          this.scrollRef = ref
        }}
        onClick={this.scrollToMyRef}
      >
        <Accordion
          expanded={this.state.isAccordionExpanded}
          className="ForgetPassword-accordion"
          header={l`Forgotten your password?`}
          accordionName="forgetPassword"
          onAccordionToggle={this.onAccordionToggle}
          noContentPadding
          noContentBorderTop={noContentBorderTop}
        >
          <ForgetPasswordForm className="ForgetPassword-form" {...props} />
        </Accordion>
      </section>
    )
  }
}

export default connect(
  (state) => ({
    forgetPwd: state.account.forgetPwd,
    forgetPasswordForm: state.forms.forgetPassword,
    isFeatureDesktopResetPasswordEnabled: isFeatureDesktopResetPasswordEnabled(
      state
    ),
  }),
  {
    forgetPwdRequest,
    resetForm,
    resetPasswordLinkRequest,
    setForgetPassword,
    setFormField,
    setFormMessage,
    touchedFormField,
  }
)(ForgetPassword)

export { ForgetPassword as WrappedForgetPassword }
