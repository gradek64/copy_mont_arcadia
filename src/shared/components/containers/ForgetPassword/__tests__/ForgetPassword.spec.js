import testComponentHelper from 'test/unit/helpers/test-component'
import { WrappedForgetPassword as ForgetPassword } from '../ForgetPassword'
import Accordion from '../../../common/Accordion/Accordion'

const renderComponent = testComponentHelper(ForgetPassword)

describe('<ForgetPasswordForm />', () => {
  const props = {}

  beforeEach(() => {
    jest.resetAllMocks()
  })

  describe('@render', () => {
    it('renders', () => {
      expect(renderComponent(props).getTree()).toMatchSnapshot()
    })

    it('renders with the provided "className"', () => {
      const { wrapper } = renderComponent({ ...props, className: 'foo' })
      expect(wrapper.prop('className')).toBe('ForgetPassword foo')
    })
  })

  describe('@lifecycle', () => {
    describe('constructor', () => {
      it('should set initial state for accordion to be contracted', () => {
        const { instance } = renderComponent(props)
        expect(instance.state.isAccordionExpanded).toBe(false)
      })
    })
  })

  describe('@events', () => {
    describe('Accordion', () => {
      describe('onAccordionToggle', () => {
        it('should toggle instance state of accordion', () => {
          const { instance, wrapper } = renderComponent(props)
          wrapper.find(Accordion).prop('onAccordionToggle')()
          expect(instance.state.isAccordionExpanded).toBe(true)
          wrapper.find(Accordion).prop('onAccordionToggle')()
          expect(instance.state.isAccordionExpanded).toBe(false)
        })
      })
    })
  })

  describe('clicking accordion', () => {
    function createDomElements({ className } = {}) {
      const containerEl = document.createElement('section')
      containerEl.getBoundingClientRect = () => ({
        top: 100,
        height: 20,
      })
      const accordionHeader = document.createElement('div')
      accordionHeader.className = className || Accordion.headerClassName
      containerEl.appendChild(accordionHeader)
      return { containerEl, accordionHeader }
    }

    const scrollY = window.scrollY
    beforeEach(() => {
      jest.spyOn(window, 'scroll').mockImplementation(() => {})
      window.scrollY = 100
    })

    afterAll(() => {
      window.scrollY = scrollY
    })

    it('scrolls the user to the forget password section when accordion is opening', () => {
      const { containerEl, accordionHeader } = createDomElements()
      const { instance, wrapper } = renderComponent(props)
      instance.scrollRef = containerEl

      wrapper
        .find('.ForgetPassword')
        .simulate('click', { target: accordionHeader })

      expect(window.scroll).toHaveBeenCalledWith({
        behavior: 'smooth',
        top: 180,
      })
    })

    it('does not scroll the page if the accordion is closing', () => {
      const { containerEl, accordionHeader } = createDomElements()
      const { instance, wrapper } = renderComponent(props)
      instance.scrollRef = containerEl

      instance.setState({ isAccordionExpanded: true })
      wrapper
        .find('.ForgetPassword')
        .simulate('click', { target: accordionHeader })

      expect(window.scroll).not.toHaveBeenCalled()
    })

    it('does not scroll the page if not clicked in accordion header', () => {
      const { containerEl } = createDomElements()
      const { instance, wrapper } = renderComponent(props)
      const someContent = document.createElement('div')
      containerEl.appendChild(someContent)
      instance.scrollRef = containerEl

      instance.setState({ isAccordionExpanded: true })
      wrapper.find('.ForgetPassword').simulate('click', { target: someContent })

      expect(window.scroll).not.toHaveBeenCalled()
    })

    it('should throw an error if the Accordion.headerClassName does not exist', () => {
      const { containerEl, accordionHeader } = createDomElements({
        className: 'Foo',
      })
      const { instance, wrapper } = renderComponent(props)
      instance.scrollRef = containerEl

      expect(() =>
        wrapper
          .find('.ForgetPassword')
          .simulate('click', { target: accordionHeader })
      ).toThrow()
    })
  })
})
