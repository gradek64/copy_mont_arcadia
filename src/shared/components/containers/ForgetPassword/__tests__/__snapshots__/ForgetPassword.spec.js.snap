// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForgetPasswordForm /> @render renders 1`] = `
<section
  className="ForgetPassword "
  onClick={[Function]}
  role="presentation"
>
  <Accordion
    accordionName="forgetPassword"
    arrowPosition="left"
    arrowStyle="primary"
    className="ForgetPassword-accordion"
    expanded={false}
    header="Forgotten your password?"
    isDisabled={false}
    noContentBorderTop={false}
    noContentPadding={true}
    noExpandedHeaderBackground={false}
    noMaxHeight={false}
    onAccordionToggle={[Function]}
    showLoader={false}
    statusIndicatorText=""
    subHeader=""
    withoutBorders={false}
  >
    <Connect(ForgetPasswordForm)
      className="ForgetPassword-form"
      enableScrollToMessage={false}
    />
  </Accordion>
</section>
`;
